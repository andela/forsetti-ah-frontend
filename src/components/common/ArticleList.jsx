import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import { formatDate } from '../../utils';
import calender from '../../assets/images/calendar.svg';

export const ArticleList = ({
  title, description, readingTime, createdAt, image, slug, history
}) => (
  <Fragment>
    <div
      className='ArticleList'
      onClick={() => history.push(`article/${slug}`)}
      role='button'
      tabIndex='-1'
      onKeyDown={null}
    >
      { createdAt
          && (
          <div className='my-2'>
            <span
              className='created font-weight-bold'
            >
              <img src={calender} alt='calender_icon' />
              <span>&nbsp; &nbsp;</span>
              { formatDate(createdAt) }
            </span>
          </div>
          )
        }
      <div
        className='card list'
      >
        <div className='row'>
          <div className={`${image ? 'col-7 col-md-8' : 'col-12'}`}>
            <div className='row py-2 py-md-3 pl-3 px-md-4'>
              <div className='col-12'>
                <h3
                  className='text-uppercase title font-weight-bold'
                >
                  { title }
                </h3>
              </div>
              <div className='col-12'>
                <p className='description' dangerouslySetInnerHTML={{ __html: description }} />
              </div>
              {
                  readingTime && (
                    <div className='col-12 mt-md-3'>
                      <h3 className='reading-time font-weight-bold'>
                        {Math.ceil(readingTime / 1000.0)}
                        {' '}
                        min read
                      </h3>
                    </div>
                  )
                }
            </div>
          </div>
          {
            image && (
              <div className='col-5 col-md-4'>
                <div
                  className='image'
                  alt={image}
                  style={{ backgroundImage: `url(${image})` }}
                />
              </div>
            )
          }
        </div>
      </div>
    </div>
  </Fragment>
);

ArticleList.defaultProps = {
  createdAt: '',
  readingTime: '',
  image: ''
};

ArticleList.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  readingTime: PropTypes.string,
  createdAt: PropTypes.string,
  image: PropTypes.string,
  slug: PropTypes.string.isRequired
};

export default withRouter(ArticleList);
