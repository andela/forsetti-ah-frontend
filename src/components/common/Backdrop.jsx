import React from 'react';

const Backdrop = ({ open }) => <div style={{ display: open ? 'block' : 'none' }} className='backdrop' />;

export default Backdrop;
