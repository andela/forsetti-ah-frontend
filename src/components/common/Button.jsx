import React from 'react';
import { Button } from 'reactstrap';

const ButtonComponent = ({
  role, onClick, type, disabled, className, children, block, outline
}) => ((
  <Button
    type={type}
    role={role}
    disabled={disabled}
    onClick={onClick}
    className={className}
    block={block}
    outline={outline}
  >
    {children}
  </Button>
));

export default ButtonComponent;
