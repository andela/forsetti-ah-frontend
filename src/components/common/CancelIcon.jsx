import React from 'react';
import cancel from '../../assets/svg/cancel.svg';

const CancelIcon = ({ clicked }) => (
  <img className='cancel-icon' src={cancel} alt='Cancel icon' onClick={clicked} />
);

export default CancelIcon;
