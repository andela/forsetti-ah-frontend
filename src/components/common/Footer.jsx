import React from 'react';
import { Row, Col } from 'reactstrap';
import VerticalListItems from './VerticalListItems';

const Footer = () => {
  const listOne = [
    { no: 1, name: 'Tech' },
    { no: 2, name: 'Philosophy' },
    { no: 3, name: 'Life' },
    { no: 4, name: 'Politics' }
  ];

  const listTwo = [
    { no: 5, name: 'Politics' },
    { no: 6, name: 'Culture' },
    { no: 7, name: 'Nature' },
    { no: 8, name: 'Music' }
  ];

  const listThree = [
    { no: 9, name: 'Finance' },
    { no: 10, name: 'Adventure' },
    { no: 11, name: 'Motivational' },
    { no: 12, name: 'Spiritual' }
  ];

  return (
    <footer className='footer'>
      <Row className='footer-upper-section'>
        <Col sm={{ size: 12 }} lg={{ size: 4 }} className='footer-logo'>
          <h6 className='footer-logo-text'>
            Authors Haven
          </h6>
        </Col>
        <Col sm={{ size: 4 }} className='footer-links'>
          <div className='footer-links-box'>
            <VerticalListItems items={listOne} className='footer-links-box-list' />
            <VerticalListItems items={listTwo} className='footer-links-box-list' />
            <VerticalListItems items={listThree} className='footer-links-box-list' />
          </div>
        </Col>
        <Col sm={{ size: 4 }} className='footer-top-logo'>
          <i className='fas fa-arrow-up footer-top-logo-icon' />
        </Col>
      </Row>
      <Row className='footer-lower-section'>
        <Col>
          <p className='footer-lower-section-text'> &copy; 2019</p>
        </Col>
      </Row>
    </footer>
  );
};

export default Footer;
