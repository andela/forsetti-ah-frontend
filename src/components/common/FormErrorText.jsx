import React from 'react';

const FormErrorText = ({ message, className }) => {
  const text = message ? (
    <p className={className}>
      {message}
    </p>
  ) : null;
  return text;
};

export default FormErrorText;
