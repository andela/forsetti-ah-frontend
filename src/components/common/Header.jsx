import React from 'react';
import { connect } from 'react-redux';
import { Button } from 'reactstrap';
import HorizontalListItems from './HorizontalListItems';
import { openModalAction } from '../../actions';
import NavBarItems from './NavBarItems';


const Header = ({ clicked, dispatch }) => {
  const openLoginModal = () => dispatch(openModalAction());
  const linkItems = [
    { no: 1, name: 'Tech' },
    { no: 2, name: 'Philosophy' },
    { no: 3, name: 'Life' },
    { no: 4, name: 'Politics' },
    { no: 5, name: 'Nature' },
    { no: 6, name: 'Music' }
  ];

  const menuItems = [];

  const isLoggedIn = () => {
    const token = localStorage.getItem('token');

    if (token) {
      return menuItems.push({ no: 1, text: 'Write Post' });
    }
    return menuItems.push({ no: 1, text: 'Sign in', onClick: openLoginModal }, { no: 2, text: 'Sign up' });
  };

  isLoggedIn();

  return (
    <nav className='header d-flex justify-content-between'>
      <div className='header-logo'>
        <h1 className='header-logo-text'>Authors Haven</h1>
      </div>
      <div className='header-links'>
        <HorizontalListItems items={linkItems} className='header-links-list' />
      </div>
      <div className='header-nav'>
        <NavBarItems menuItems={menuItems} className='header-nav-menu' />
        <Button onClick={clicked} className='header-nav-hamburger'>
          <i className='fas fa-bars' />
        </Button>
      </div>
    </nav>
  );
};

const HeaderComponent = connect()(Header);

export {
  HeaderComponent,
  Header,
};
