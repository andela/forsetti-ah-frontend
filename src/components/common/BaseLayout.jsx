import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import { Row, Col } from 'reactstrap';
import { SideDrawerComponent } from '../SideDrawer';
import { HeaderComponent } from './Header';
import Footer from './Footer';
import LoginModal from '../LoginModal';
import { LandingPageComponent } from '../../containers/LandingPage';
import NotFound from '../NotFound';
import Article from '../../containers/Article';
import { showSideDrawerAction, hideSideDrawerAction } from '../../actions';
import { AuthSocialComponent } from '../AuthSocial';

const BaseLayoutComponent = ({ showSideDrawer, modal, dispatch }) => {
  const showSideDrawerHandler = () => {
    dispatch(showSideDrawerAction());
  };

  const hideSideDrawerHandler = () => {
    dispatch(hideSideDrawerAction());
  };
  const { showSideDrawer: showSideDraw } = showSideDrawer;
  const { showModal } = modal;
  return (
    <Router>
      <SideDrawerComponent show={showSideDraw} closed={hideSideDrawerHandler} />
      <LoginModal isOpen={showModal} />
      <HeaderComponent clicked={showSideDrawerHandler} />
      <Switch>
        <Route path='/' component={LandingPageComponent} exact />
        <Route exact path='/article/:slug' component={Article} />
        <Route path='/auth/social' component={AuthSocialComponent} />
        <Route component={NotFound} />
      </Switch>
      <div className='footer'>
        <Row>
          <Col md={{ size: 12 }}>
            <Footer />
          </Col>
        </Row>
      </div>
    </Router>
  );
};

const mapStateToProps = (state) => {
  const { showSideDrawer, modal } = state;
  return { showSideDrawer, modal };
};

const BaseLayout = connect(mapStateToProps)(BaseLayoutComponent);

export {
  BaseLayout,
  BaseLayoutComponent
};
