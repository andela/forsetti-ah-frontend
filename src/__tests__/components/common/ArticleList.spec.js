import React from 'react';
import { shallow } from 'enzyme';
import { ArticleList } from '../../../components/common/ArticleList';

const props = {
  title: 'I am a very good boy',
  description: 'I dont know what to say like seriously',
  readingTime: '0.03',
  createdAt: '2019-05-15T10:59:12.703Z',
  image: 'https://res.cloudinary.com/forsetti/image/upload/v1557849271/forsetti/s1tujji64ai6wrdc3h93.jpg',
  slug: 'i-am-a-very-good-boy-1233343',
  dispatch: jest.fn(),
  history: {
    push: jest.fn()
  }
};
describe('App', () => {
  test('Should render component successfully', () => {
    const wrapper = shallow(<ArticleList { ...props } />);
    expect(wrapper.length).toEqual(1);
  });
  
  test('should', () => {
    const wrapper = shallow(<ArticleList {...props} />);

    wrapper.find('.ArticleList').simulate('click', {
      history: () => {}
    });
  });
  test('Should maintain existing snapshot', () => {
    const wrapper = shallow(<ArticleList {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
