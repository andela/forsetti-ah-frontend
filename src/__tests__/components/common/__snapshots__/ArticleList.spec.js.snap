// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should maintain existing snapshot 1`] = `ShallowWrapper {}`;
